flink-connector-jdbc-core/src/main/java/org/apache/flink/connector/jdbc/core/database/catalog/JdbcCatalogTypeMapper.java

```diff
@@ -16,22 +16,19 @@
  * limitations under the License.
  */
 
-package org.apache.flink.connector.jdbc.databases.db2.dialect;
+package org.apache.flink.connector.jdbc.core.database.catalog;
 
-import org.apache.flink.annotation.Internal;
-import org.apache.flink.connector.jdbc.dialect.JdbcDialect;
-import org.apache.flink.connector.jdbc.dialect.JdbcDialectFactory;
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.table.catalog.ObjectPath;
+import org.apache.flink.table.types.DataType;
 
-/** Factory for {@link Db2Dialect}. */
-@Internal
-public class Db2DialectFactory implements JdbcDialectFactory {
-    @Override
-    public boolean acceptsURL(String url) {
-        return url.startsWith("jdbc:db2:");
-    }
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
 
-    @Override
-    public JdbcDialect create() {
-        return new Db2Dialect();
-    }
+/** Separate the jdbc meta-information type to flink table type into the interface. */
+@PublicEvolving
+public interface JdbcCatalogTypeMapper {
```

Comment:
How about rename to 'JdbcTypeMapper' ?

Reply:
we use the same pattern on all Jdbc{kink}Anything, where kind is source, sink, dialect, catalog (already from before the refactor)

this was called "JdbcDialectTypeMapper"